### PR TITLE
Columns property in the px-data-grid

### PR DESCRIPTION
**Pull request by Toshegg**
Added columns property that allows to pass custom columns to
px-data-grid.

### CODE REVIEW

**Comment by alump** [comment]
Sorry I rebased this to master, that's why forced push.

**Comment by alump** [comment]
And yeah, I used the observer trick to pair the data column object to HTML element column. Allowing us to do nice value updating in the future (observer can just update values like hidden, frozen on JS object when those change on DOM element). So now column.element should return the element, if that dom-repeat magic is ran.

**Comment by alump** [comment]
Did magic squeeze of my 4 commits, so now it's just one commit. More commits might still happen, if/when issues found.

**Comment by Toshegg** [comment]
- Need to throw warning at least if path isn't passed, because user can define column.name as 'First Name', for example, since header is not required parameter, what can cause errors.
- @alump, why did you change switch/case to if/else? Switch/case construction is more readable in this case, and if you want "cell" to be the default option, switch/case contains "default" keyword.

File: px-data-grid.html [hunk 1]
```diff
@@ -417,8 +423,16 @@
           );
         }
 
+        _readValue(column, item) {
```
Comment (Toshegg) [inline px-data-grid.html]: @alump, We already have _getValue method, can it be used for your purposes?

Comment (alump) [reply px-data-grid.html]: _readValue has extra functionality required for now (as far as items are used). But _getItems usage can be replaced with _readValue usage. And if this is done, it's ok to rename _readValue to _getValue.

**Pull request merged**